core/dbt/contracts/graph/components.py

```diff
@@ -0,0 +1,53 @@
+from dataclasses import dataclass, field
+from datetime import timedelta
+from dbt.artifacts.resources import (
+    FreshnessThreshold as FreshnessThresholdResource,
+    HasRelationMetadata as HasRelationMetadataResource,
+    Time as TimeResource,
+)
+from typing import Dict, Optional
+
+
```

Comment:
Reading through these, I'm in favor of defining the classes below in dbt/artifacts. Their methods are fairly slim and minimal, so I don't think it'd add much to the interface burden of dbt-artifacts to maintain for downstream consumers. None of them bring in dependencies on core and are fairly simple definitions based on the raw data. 

It would simplify things a good deal too instead of having this inheritance pattern for fairly intermediate components of reosurces.